### PR TITLE
Fixes #16839 - sort by origin is possible

### PR DESCRIPTION
**Pull request by lizagilman**

### CODE REVIEW

**Comment by tbrisker** [comment]
add to PR whitelist

File: test/factories/facts_related.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 FactoryGirl.define do
   factory :fact_name do
     sequence(:name) {|n| "fact#{n}" }
+    type 'PuppetFactName'
```
Comment (ohadlevy) [inline test/factories/facts_related.rb]: @lizagilman I would suggest to try without this line, as it turns out break many other tests.

File: app/controllers/fact_values_controller.rb [hunk 2]
```diff
@@ -5,13 +5,7 @@ class FactValuesController < ApplicationController
   before_action :setup_search_options, :only => :index
 
   def index
-    base = resource_base
-    begin
-      values = base.my_facts.search_for(params[:search], :order => params[:order])
-    rescue => e
```
Comment (dLobatog) [inline app/controllers/fact_values_controller.rb:11]: In my opinion this error handling removal is OK - it's from 2011 and I'd prefer not to hide potential exceptions

File: app/models/fact_value.rb [hunk 3]
```diff
@@ -15,6 +15,7 @@ class FactValue < ActiveRecord::Base
   scoped_search :relation => :host,      :on => :name, :complete_value => true, :rename => :host, :ext_method => :search_by_host_or_hostgroup, :only_explicit => true
   scoped_search :relation => :hostgroup, :on => :name, :complete_value => true, :rename => :"host.hostgroup", :ext_method => :search_by_host_or_hostgroup, :only_explicit => true
   scoped_search :relation => :fact_name, :on => :short_name, :complete_value => true, :aliases => ["fact_short_name"]
+  scoped_search :relation => :fact_name, :on => :type, :complete_value => false, :only_explicit => true, :aliases => ["origin"]
```
Comment (dLobatog) [inline app/models/fact_value.rb:18]: Why `complete_value` is false? There are only 3 or 4 options in a typical installation, which makes it a good candidate for autocompletion IMO. Same for `only_explicit`, why?

**Review by dLobatog — CHANGES_REQUESTED** [review]
Works fairly well, thanks @lizagilman - I left a couple of comments about the scoped_search options.

File: app/models/fact_value.rb (hunk 3 above)
Comment (ohadlevy) [reply app/models/fact_value.rb:18]: The type field value is not user friendly, the initial bug was with sort, fixing the search imho is out of scope for this pr?

Comment (dLobatog) [reply app/models/fact_value.rb:18]: Oh I see it now - "type = ForemanAnsible::FactName" is definitely bad... this is a good candidate for `ext_method` though.

**Review by dLobatog — APPROVED** [review]

**Comment by dLobatog** [comment]
Merging after seeing the type values are things like "ForemanAnsible::FactName" - which shouldn't be displayed to the user. This should use `ext_method` or something similar to avoid this problem.

**Comment by dLobatog** [comment]
Thanks @lizagilman @ohadlevy

**Pull request merged**